pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -268,8 +269,38 @@ func newVersionedResources(rs []ctlres.Resource) versionedResources {
 		// (Annotations may have been copied from versioned resources
 		// onto transient resources for non-versioning related purposes).
 		_, hasVersionedAnn := res.Annotations()[versionedResAnnKey]
-		if hasVersionedAnn && !res.Transient() {
+		_, hasVersionedOrigAnn := res.Annotations()[versionedResOrigAnnKey]
+
+		if hasVersionedAnn {
 			result.Versioned = append(result.Versioned, res)
+			if hasVersionedOrigAnn {
+				result.NonVersioned = append(result.NonVersioned, res.DeepCopy())
+			}
+		} else {
+			result.NonVersioned = append(result.NonVersioned, res)
+		}
+	}
+	return result
+}
+
+func existingVersionedResources(rs []ctlres.Resource) versionedResources {
+	var result versionedResources
+	for _, res := range rs {
+		// Expect that versioned resources should not be transient
+		// (Annotations may have been copied from versioned resources
+		// onto transient resources for non-versioning related purposes).
+		_, hasVersionedAnn := res.Annotations()[versionedResAnnKey]
+		_, hasVersionedOrigAnn := res.Annotations()[versionedResOrigAnnKey]
+
+		versionedRs := VersionedResource{res: res, allRules: nil}
+		_, versionExists := versionedRs.BaseNameAndVersion()
```

Comment:
nit: versionExists -> version. it's actually a version thats parsed, not a bool.